docs/design/design_trivy_file_system_scanner.md

```diff
@@ -0,0 +1,200 @@
+- In this mode, we don't need imagePullSecret/any secret to pass to scan job
+- ImagePullPolicy should be set as Never
+- nodeName should be configured explicitly. 
+
+```yaml
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: scan-nginx-with-trivy
+  namespace: starboard-operator
+spec:
+  backoffLimit: 0
+  template:
+    spec:
+      restartPolicy: Never
+      # Explicit nodeName indicates our intention to schedule a scan pod
+      # on the same node where the nginx workload is running.
+      # This could also imply considering taints and tolerations and other
+      # properties respected by K8s scheduler.
+      nodeName: kind-control-plane
+      volumes:
+        - name: scan-volume
+          emptyDir: { }
+      initContainers:
+        # The trivy-get-binary init container is used to copy out the trivy executable
+        # binary from the upstream Trivy container image, i.e. aquasec/trivy:0.19.2,
+        # to a shared emptyDir volume.
+        - name: trivy-get-binary
+          image: aquasec/trivy:0.19.2
+          command:
+            - cp
+            - -v
+            - /usr/local/bin/trivy
+            - /var/starboard/trivy
+          volumeMounts:
+            - name: scan-volume
+              mountPath: /var/starboard
+         # The trivy-download-db container is using trivy executable binary
+         # from the previous step to download Trivy vulnerability database
+         # from GitHub releases page.
+        - name: trivy-download-db
+          image: aquasec/trivy:0.19.2
+          command:
+            - /var/starboard/trivy
+            - --download-db-only
+            - --cache-dir
+            - /var/starboard/trivy-db
+          volumeMounts:
+            - name: scan-volume
+              mountPath: /var/starboard
+      containers:
+        # The scan-nginx container is based on the container image that
+        # we want to scan with Trivy. However, it has overwritten command (entrypoint)
+        # to invoke trivy file system scan. The scan results are output to stdout
+        # in JSON format so we can parse them and store as VulnerabiltyReport instance.
+        - name: scan-nginx
+          image: example.registry.com/nginx:1.16
+          # To scan image layers cached on K8s node without pulling
+          # it from a remote registry.
+          imagePullPolicy: Never
+          command:
+            - /var/starboard/trivy
+            - --cache-dir
+            - /var/starboard/trivy-db
+            - fs
+            - --format
+            - json
+            - --ignore-unfixed
+            - /
+          volumeMounts:
+            - name: scan-volume
+              mountPath: /var/starboard
+```
+
+##### Current Limitations for this option:
+1. If controller is created, and it doesn’t have any active pods then this mode will not scan it
+2. We cannot identify pods running for cronjob, so those will not be scanned.
```

Comment:
Can you elaborate on why we cannot identify pods running for CronJobs? Is it because we scan by K8s controller not by pod?

Reply:
yes, boz when starboard receives event of cronjob we may not have an active pod running for it. and also getting active pods of cron job is not straight forward way, so added it as limitation for now.
this is one way to get jobs for cronjob, later we will have to get active pods for these jobs. https://github.com/kubernetes/kubernetes/blob/e640a01219736dd051dc3a9b8ac4630ec18b6b25/pkg/controller/cronjob/cronjob_controllerv2.go#L226